app/controllers/saml_idp_controller.rb

```diff
@@ -148,10 +148,24 @@ def saml_response
     encode_response(
       current_user,
       authn_context_classref: requested_authn_context,
-      reference_id: auth_identity.session_uuid.gsub(/^_/, '')
+      reference_id: auth_identity.session_uuid.gsub(/^_/, ''),
+      encryption: saml_response_encryption_opts
     )
   end
 
+  def saml_response_encryption_opts
+    encryption_opts = nil
+    if current_user.last_identity
+      sp = ServiceProvider.new(current_user.last_identity.service_provider)
```

Comment:
Good question. I wasn't sure how best to identify the service provider for the current request.

It looks like `auth_identity` is for *setting* the current user's auth identity. In this case I want to fetch it only.

Should I maybe use `saml_request.service_provider.identifier` with `ServiceProvider.find` instead?